lib/pure/strmisc.nim

```diff
@@ -85,6 +86,76 @@ proc rpartition*(s: string, sep: string): (string, string, string)
 
   return partition(s, sep, right = true)
 
+
+func parseFloatThousandSep*(str: openArray[char]; sep = ','; decimalDot = '.'): float {.since: (1, 5).} =
+  ## Convenience func for `parseFloat` which allows for thousand separators,
+  ## this is designed to parse floats as found in the wild formatted for humans.
+  ##
+  ## The following assumptions and requirements must be met:
+  ## - `str` must not be empty.
+  ## - `str` must be stripped of trailing and leading whitespaces.
+  ## - `sep` must not be `'-'`.
+  ## - `decimalDot` must not be `'-'` nor `' '`.
+  ## - `sep` and `decimalDot` must be different.
+  ## - Exactly 1 separator must appear after each 3 consecutive inner digits to the left of the dot and nowhere else.
+  ##
+  ## See also:
+  ## * `parseFloat <strutils.html#parseFloat,string>`_
+  runnableExamples:
+    doAssert parseFloatThousandSep("-1,000") == -1000.0
+    doAssert parseFloatThousandSep("10,000.000") == 10000.0
+    doAssert parseFloatThousandSep("1,000,000.000") == 1000000.0
+    doAssert parseFloatThousandSep("10,000,000.000") == 10000000.0
+    doAssert parseFloatThousandSep("10.000,0", '.', ',') == 10000.0
+    doAssert parseFloatThousandSep("1'000'000,000", '\'', ',') == 1000000.0
+    doAssert parseFloatThousandSep("1000,000") == 1000000.0
```

Comment:
add a comment here, it'll avoid a lengthy/hard to read description at the top:
```nim
you can omit `sep`, but then all subsequent `sep` to the left must also be omitted.
```

Reply:
As said above, this example shouldn't work.